be/src/util/bthreads/semaphore.h

```diff
@@ -16,6 +16,8 @@
 
 #include <bthread/butex.h>
 #include <bthread/processor.h> // cpu_relax()
+// resolve `barrier` macro conflicts with boost/thread.hpp header file
+#undef barrier
```

Comment:
Use this macro where a conflict arises.

Reply:
yes, it was intentionally  introduced to resolve conflicts